### ISSUE TITLE
[BUG] Puppeting from an EvMenu results in a lack of Exit CommandSet

### ISSUE THREAD

**Issue by Exsavio** [evennia/evennia#2855]
#### Describe the bug
When account.puppet_obj(session, character) is invoked from a Session level EvMenu, the character is puppeted without error however the Character does not receive an Exit Command Sets, and thus cannot move.

#### To Reproduce
1. Create an Account and a Character (the initial superuser suffices)
2. Create an EvMenu: 
```
def start(caller, raw_string, **kwargs):
	
	text = "Press any button to load first character"
	options = [{ "key"  : "_default", "desc" : "", "goto" : "play" },]

	return text, options

def play(caller, raw_string, **kwargs):

	session = caller
	account = session.account
	character = account.db._playable_characters[0]

	#code taken from default CmdIC command:
	try:
		account.puppet_object(session, character)
		account.db._last_puppet = character
		logger.log_sec(
			"Puppet Success: (Caller: %s, Target: %s, IP: %s)."
			% (account, character, session.address)
		)
	except RuntimeError as exc:
		self.msg("|rYou cannot become |C%s|n: %s" % (character.name, exc))
		logger.log_sec(
			"Puppet Failed: %s (Caller: %s, Target: %s, IP: %s)."
			% (exc, account, character, session.address)
		)

	return None, None
```
3.  Override DefaultAccount in order to trigger the menu on login (using at_look())
```
        if target and not is_iter(target):
            # single target - just show it
            if hasattr(target, "return_appearance"):
                return target.return_appearance(self)
            else:
                return f"{target} has no in-game appearance."
        else:
            EvMenu(session, "account.menu_account",
                auto_look=False, auto_quit=False, auto_help=False,
                persistent=False, cmd_on_exit = None, session=session,
                cmdset_mergetype="Replace")

            return ""
```

4. Reload
5. Logout or OOC
6. Press any button
7. Observe inability to move, as Exit Command Sets not available to Character.
8. Reload
9. Observe ability to move restored.

#### Expected behavior
Ability to utilize movement commands upon puppeting from an EvMenu

#### Environment, Evennia version, OS etc
```
    Evennia 1.0-dev (rev 77171aee8a4) (rev 77171aee8a4)
    OS: nt
    Python: 3.10.5
    Twisted: 21.7.0
    Django: 4.0.5
```

#### Additional context
When establishing a simplified version of this issue to reproduce,  it was observed that if the EvMenu had no options (i.e. start retuned None, None) the issue did not reoccur.

**Comment by Griatch** [comment]
A rather confusing error, this :confused: An initial investigation of it didn't show why this could be. Would need more time to look at it.

**Comment by Griatch** [comment]
This was a tricky one; and it is actually completely unrelated to puppeting. It turns out the Account cmdset overloaded with the menu cmdset (despite the menu getting started on the Session) due to a fallback in the cmdhandler that I've now removed. The result was that the menu cmdset on the Account was not removed when the menu closed, but hung around - and this cmdset ignores objects/exits and has a higher prio, so it would just skip all exits. 

Fixed now; we'll see if removing the fallback has any other effects, but at least nothing the unittests can catch.
